src/logging.h

```diff
@@ -173,17 +192,31 @@ static inline void LogPrintf(const char* fmt, const Args&... args)
             /* Original format string will have newline so don't add one here */
             log_msg = "Error \"" + std::string(fmterr.what()) + "\" while formatting log message: " + fmt;
         }
-        LogInstance().LogPrintStr(log_msg);
+        const SourceLocation source_location = std::make_pair(source_file, source_line);
+        LogInstance().LogPrintStr(log_msg, logging_function, source_location, skip_disk_usage_rate_limiting);
     }
 }
 
+// Unconditional logging. Uses basic rate limiting to mitigate disk filling attacks.
+#define LogPrintf(...) LogPrintf_(__func__, __FILE__, __LINE__, /* skip_disk_usage_rate_limiting */ false, __VA_ARGS__)
+
+// Unconditional logging WITHOUT rate limiting. Use only for log messages that
+// MUST NOT be rate limited no matter how often they are logged. That requirement
+// should be extremely rare, so please use with care. Prefer LogPrintf(...) if
+// possible.
+#define LogPrintfWithoutRateLimiting(...) LogPrintf_(__func__, __FILE__, __LINE__, /* skip_disk_usage_rate_limiting */ true, __VA_ARGS__)
```

Comment:
This name is really awkward. I would prefer a more compact name, but don't really have any good suggestions.

Reply:
Heh, I agree :)  Naming is hard! I'm happy to change the name of course.

I expect `LogPrintfWithoutRateLimiting` to be used once: for `UpdateTip` which is the only non-category log message which we ever expect to legitimately log at a rate of >1 MB/hour (or whatever hourly source-location threshold we choose).

One alternative would be to call `LogPrintf_` directly from `UpdateTip` as it is a one-off. That would solve the naming problem too :)